QWeb/internal/xhr.py

```
@@ -16,12 +16,256 @@
+  window.__xhrMon = Object.assign(window.__xhrMon || {}, {
+    installed: true,
+    pending: (window.__xhrMon && typeof window.__xhrMon.pending === "number") ? window.__xhrMon.pending : 0,
+    fetchPatched: false,
+    xhrPatched: false,
+    lastMutationTs: Date.now(),
+    observerStarted: false
+  });
+
+  try {
+    if (!window.__xhrMon.fetchPatched && typeof window.fetch === "function") {
+      const _orig = window.fetch;
+      window.fetch = function() {
+        try { window.__xhrMon.pending++; } catch(e) {}
+        const p = _orig.apply(this, arguments);
+        const dec = function(){ try { window.__xhrMon.pending--; } catch(e) {} };
+        return p && typeof p.finally === "function"
+          ? p.finally(dec)
+          : p.then(function(r){ dec(); return r; }, function(e){ dec(); throw e; });
+      };
+      window.__xhrMon.fetchPatched = true;
+    }
+  } catch(e){}
+
+  try {
+    if (!window.__xhrMon.xhrPatched && window.XMLHttpRequest && window.XMLHttpRequest.prototype) {
+      const _open = XMLHttpRequest.prototype.open;
+      const _send = XMLHttpRequest.prototype.send;
+      XMLHttpRequest.prototype.open = function() { this.__xhrTracked = true; return _open.apply(this, arguments); };
+      XMLHttpRequest.prototype.send = function() {
+        if (this.__xhrTracked) {
+          try { window.__xhrMon.pending++; } catch(e) {}
+          this.addEventListener("loadend", function(){ try { window.__xhrMon.pending--; } catch(e) {} }, { once:true });
+        }
+        return _send.apply(this, arguments);
+      };
+      window.__xhrMon.xhrPatched = true;
+    }
+  } catch(e){}
+
+  try {
+    if (!window.__xhrMon.observerStarted && window.MutationObserver) {
+      const obs = new MutationObserver(function(){ window.__xhrMon.lastMutationTs = Date.now(); });
+      obs.observe(document.documentElement || document.body, {
+        childList:true, subtree:true
+      });
+      window.__xhrMon.observerStarted = true;
+    }
+  } catch(e){}
```

Silent error discarding of all errors seems bit questionable.